From: Takashi Iwai <tiwai@suse.de>
Subject: Fix kABI breakage of iio_buffer
Patch-mainline: Never, kABI fix
References: bsc#1051510

The recent iio fix
  patches.drivers/iio-buffer-make-length-types-match-kfifo-types
caused a kABI breakage due to the changes from int to unsigned and
size_t.  For fixing this, revert to int for declaration, and use only
unsigned for keeping alignment.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 include/linux/iio/buffer_impl.h |   12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

--- a/include/linux/iio/buffer_impl.h
+++ b/include/linux/iio/buffer_impl.h
@@ -52,8 +52,11 @@ struct iio_buffer_access_funcs {
 	int (*request_update)(struct iio_buffer *buffer);
 
 	int (*set_bytes_per_datum)(struct iio_buffer *buffer, size_t bpd);
+#ifdef __GENKSYMS__
+	int (*set_length)(struct iio_buffer *buffer, int length);
+#else
 	int (*set_length)(struct iio_buffer *buffer, unsigned int length);
-
+#endif
 	int (*enable)(struct iio_buffer *buffer, struct iio_dev *indio_dev);
 	int (*disable)(struct iio_buffer *buffer, struct iio_dev *indio_dev);
 
@@ -70,11 +73,16 @@ struct iio_buffer_access_funcs {
  * those writing new buffer implementations.
  */
 struct iio_buffer {
+#ifdef __GENKSYMS__
+	int length;
+	int bytes_per_datum;
+#else
 	/** @length: Number of datums in buffer. */
 	unsigned int length;
 
 	/**  @bytes_per_datum: Size of individual datum including timestamp. */
-	size_t bytes_per_datum;
+	unsigned int bytes_per_datum;
+#endif
 
 	/**
 	 * @access: Buffer access functions associated with the
